data/txHandler.go

```diff
@@ -294,16 +294,11 @@ func (handler *TxHandler) backlogWorker() {
 				// this is never happening since handler.backlogQueue is never closed
 				return
 			}
-			if wi.capguard != nil {
-				if err := wi.capguard.Release(); err != nil {
-					logging.Base().Warnf("Failed to release capacity to ElasticRateLimiter: %v", err)
-				}
-			}
-			if handler.checkAlreadyCommitted(wi) {
+			ac := handler.checkAlreadyCommitted(wi)
+			// mark the capacity as released, and as served too, if it's alreadyCommitted
+			wi.capguard.HandleCapguard()
```

Comment:
this is not functionally equivalent to the original code: for each incoming `wi` Released is called in order to release capacity for a client. Then, Served called after some delay when `wi` gets actually queued into the verifier, so it could account for delays in batch verifier.
But I think this Served is misplaced and should be in `case wi, ok := <-handler.postVerificationQueue:` branch when we actually get `wi` verified by the batch signature verification.
@AlgoAxel @algonautshant opinions?